crates/blockchain/mempool.rs

```diff
@@ -70,10 +82,23 @@ impl Mempool {
             .inner
             .write()
             .map_err(|error| StoreError::MempoolWriteLock(error.to_string()))?;
+
+        // Prune the order queue if it has grown too much
+        if inner.txs_order.len() > MEMPOOL_PRUNE_THRESHOLD {
+            let mut new_txs_order = VecDeque::with_capacity(MEMPOOL_MAX_SIZE * 2);
+            for tx in inner.txs_order.iter() {
+                if inner.transaction_pool.contains_key(tx) {
+                    new_txs_order.push_back(*tx);
+                }
+            }
+            inner.txs_order = new_txs_order;
```

Comment:
Can we use [`retain`](https://doc.rust-lang.org/std/collections/struct.VecDeque.html#method.retain) here?


```suggestion
            inner.txs_order.retain(|tx| inner.transaction_pool.contains_key(tx));
```

Reply:
I tried to, but i can't because of borrowing inner both mutably and inmutably